.github/workflows/main.yml

```diff
@@ -0,0 +1,285 @@
+        id: include
+        run: |
+          # Determine which platform jobs to run
+
+          # Returns 'true' if the input platform list matches any of the platform monikers given as argument,
+          # 'false' otherwise.
+          # arg $1: platform name or names to look for
+          function check_platform() {
+            if [[ '${{ !secrets.JDK_SUBMIT_FILTER || startsWith(github.ref, 'refs/heads/submit/') }}' == 'false' ]]; then
+              # If JDK_SUBMIT_FILTER is set, and this is not a "submit/" branch, don't run anything
+              echo 'false'
+              return
+            fi
+
+            if [[ $GITHUB_EVENT_NAME == workflow_dispatch ]]; then
+              input='${{ github.event.inputs.platforms }}'
+            elif [[ $GITHUB_EVENT_NAME == push ]]; then
+              input='${{ secrets.JDK_SUBMIT_PLATFORMS }}'
+            else
+              echo 'Internal error in GHA'
+              exit 1
+            fi
+
+            normalized_input="$(echo ,$input, | tr -d ' ')"
+            if [[ "$normalized_input" == ",," ]]; then
+              # For an empty input, assume all platforms should run
+              echo 'true'
+              return
+            else
+              # Check for all acceptable platform names
+              for part in $* ; do
+                if echo "$normalized_input" | grep -q -e ",$part," ; then
+                  echo 'true'
+                  return
+                fi
+              done
+            fi
+
+            echo 'false'
+          }
+
+          echo "::set-output name=linux-x64::$(check_platform linux-x64 linux x64)"
+          echo "::set-output name=linux-x86::$(check_platform linux-x86 linux x86)"
+          echo "::set-output name=linux-x64-variants::$(check_platform linux-x64-variants variants)"
+          echo "::set-output name=linux-cross-compile::$(check_platform linux-cross-compile cross-compile)"
+          echo "::set-output name=macos-x64::$(check_platform macos-x64 macos x64)"
+          echo "::set-output name=macos-aarch64::$(check_platform macos-aarch64 macos aarch64)"
+          echo "::set-output name=windows-x64::$(check_platform windows-x64 windows x64)"
+          echo "::set-output name=windows-aarch64::$(check_platform windows-aarch64 windows aarch64)"
+
+  ###
+  ### Build jobs
+  ###
+
+  build-linux-x64:
+    name: linux-x64
+    needs: select
+    uses: ./.github/workflows/build-linux.yml
+    with:
+      platform: linux-x64
+      apt-gcc-version: '10=10.3.0-1ubuntu1~20.04'
+    # The linux-x64 jdk bundle is used as buildjdk for the cross-compile job
+    if: needs.select.outputs.linux-x64 == 'true' || needs.select.outputs.linux-cross-compile == 'true'
+
+  build-linux-x86:
+    name: linux-x86
+    needs: select
+    uses: ./.github/workflows/build-linux.yml
+    with:
+      platform: linux-x86
+      apt-gcc-version: '10-multilib'
+      apt-architecture: ':i386'
+      apt-add-architecture: 'sudo dpkg --add-architecture i386'
```

Comment:
Feels a bit weird to have the entire command line here. I'd expect to see something like `apt-add-architecture: i386`.

Reply:
If I do that, there need to be some kind of if statement in the called workflow, since if that input argument is left out, we'd get a command line like `sudo dpkg --add-architecture` which I assume is illegal syntax (or, possibly worse, does something other than a no-op).

I think there can be room for additional improvement in especially the "special" linux builds, but I had to draw the line somewhere, to be able to finish this PR.

Reply:
I think you can make a conditionalized step that runs in case the value is not empty, and skip the step otherwise. If that is hard, it can be done in the followup, sure.

Reply:
I cleaned up the entire "apt architecture" handling a bit. Some part of the code got more complex, but overall I think it was worth it due to the higher level of abstraction.